lightningd/peer_control.c

```diff
@@ -2975,4 +2976,78 @@ static const struct json_command dev_reenable_commit = {
 	"Re-enable the commit timer on peer {id}"
 };
 AUTODATA(json_command, &dev_reenable_commit);
+
+struct dev_forget_channel_cmd {
+	struct short_channel_id scid;
+	struct peer *peer;
+	bool force;
+	struct command *cmd;
+};
+
+static void process_dev_forget_channel(struct bitcoind *bitcoind UNUSED,
+				       const struct bitcoin_tx_output *txout,
+				       void *arg)
+{
+	struct json_result *response;
+	struct dev_forget_channel_cmd *forget = arg;
+	if (txout != NULL && !forget->force) {
+		command_fail(forget->cmd,
+			     "Cowardly refusing to forget channel with an "
+			     "unspent funding output, if you know what "
+			     "you're doing you can override with "
+			     "`force=true`, otherwise consider `close` or "
+			     "`dev-fail`! If you force and the channel "
+			     "confirms we will not track the funds in the "
+			     "channel");
+		return;
+	}
+	free_peer(forget->peer, "dev-forget-channel called");
+	if (peer_persists(forget->peer))
+		wallet_channel_delete(forget->cmd->ld->wallet, forget->peer->channel->id);
+
+	response = new_json_result(forget->cmd);
+	json_object_start(response, NULL);
+	json_object_end(response);
+	command_success(forget->cmd, response);
+
+}
+
+static void json_dev_forget_channel(struct command *cmd, const char *buffer,
+				    const jsmntok_t *params)
+{
+	jsmntok_t *nodeidtok, *forcetok;
+	struct dev_forget_channel_cmd *forget = tal(cmd, struct dev_forget_channel_cmd);
+	forget->cmd = cmd;
+	if (!json_get_params(cmd, buffer, params,
+			     "node_id", &nodeidtok,
```

Comment:
Trivial: let's use 'id' as the name here, which is what we standardized on everywhere else.